vendor/github.com/OpenBazaar/multiwallet/service/wallet_service.go

```diff
@@ -522,7 +522,7 @@ func (ws *WalletService) saveSingleTxToDB(u model.Transaction, chainHeight int32
 	cb.Value = value
 	cb.WatchOnly = (hits == 0)
 	saved, err := ws.db.Txns().Get(*txHash)
-	if err != nil {
+	if err != nil || saved.WatchOnly != cb.WatchOnly {
```

Comment:
This seems to fix this one place, but will allow our bad data to infect at least two other places:

1. vendor/github.com/OpenBazaar/multiwallet/util/balance.go within `CalcBalance` which omits `WatchOnly` txs so the balance will be misrepresented. The impact is incorrect balances from `/wallet/balance`.
2. repo/db/txns.go within `GetAll` which flags whether certain records should or shouldn't be returned for downstream usage. The impact is missing transactions from `/wallet/transactions`.

We either need to handle the two above cases or we need to reconsider the data migration approach.

Reply:
Okay, we can disregard. I misunderstood where this check was being added and arrived at the wrong conclusion. I do note that this will update the transaction record with a freshly calculated state.